micro_sam/instance_segmentation.py

```diff
@@ -1139,3 +1143,145 @@ def get_amg(
         segmenter = TiledInstanceSegmentationWithDecoder(predictor, decoder, **kwargs) if is_tiled else\
             InstanceSegmentationWithDecoder(predictor, decoder, **kwargs)
     return segmenter
+
+
+def automatic_instance_segmentation(
+    input_path: Union[os.PathLike, str],
+    output_path: Union[os.PathLike, str],
+    embedding_path: Optional[Union[os.PathLike, str]] = None,
+    model_type: str = util._DEFAULT_MODEL,
+    checkpoint_path: Optional[Union[os.PathLike, str]] = None,
+    key: Optional[str] = None,
+    ndim: Optional[int] = None,
```

Comment:
The current code will only work for 2d images. So it doesn't really make sense to expose `ndim` here.
We have two options: don't expose `ndim` or extend the function that we also support 3d segmentation by using functionality from `multi_dimensional_segmentation`.  We have to move this to a different file if we want to go with the latter option to avoid circular imports.

Reply:
Thanks for the feedback!

I went ahead and added support for 3d segmentation. However, I spotted the [TODO](https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/multi_dimensional_segmentation.py#L356) for tiling-based prediction, which is probably not fully implemented. I'll leave the comment open so that we discuss this tomorrow.

Reply:
Ah, it does work. We had to the changes I made already in the code, and that's pretty much it (tested it just now on `lucchi_test.h5` with tile 512 and halo 128 - looks as expected).